### ISSUE TITLE
Slow performance on aarch64 + GPU compared to x86 CPU

### ISSUE THREAD

**Issue by eqikkwkp25-cyber** [janvainer/speedyspeech#11]
On x86 CPU without CUDA inference is about 4 times faster than realtime with a remarkable audio quality, congratulations. 

But speedyspeech runs on a Jetson Nano, i.e. aarch64 architecture + GPUs, about 2 times slower than realtime which was unexpected. I used Python 3.6 and Torch 1.6 to be able to run it, otherwise i sticked to the requirements. CPUs seem to be very active in the beginning (loading models), RAM is sufficient (no swapping) and GPUs are used well during inference.

Should i expect a slow performance on a mobile architecture like the above?Are there any known bottle necks?

**Comment by janvainer** [comment]
Hi, thanks for testing SpeedySpeech  on Jetson Nano. I haven't personally tested the model on other architectures so this is interesting news. Regarding bottlenecks, I am not aware of any direct ones. I assume that exporting the network with TensorRT could provide substantial speedup on JetsonNano as is discussed [here](https://discuss.pytorch.org/t/optimizing-pytorch-model-for-single-board-computers-jetson-nano/81849). I am planning to do some refactoring and deployment optimization soon so stay tuned for updates. Contributions would be welcome too! :)

**Comment by eqikkwkp25-cyber** [comment]
Thank you for the hint regarding TensorRT conversion. Looking forward for your refactoring and deployment optimizations though i think latter was straightforward. 

I observed unstable audio generation towards the end of longer sentences. Is this known or should i open an issue for this?

**Comment by janvainer** [comment]
Could you post an example audio with the unstable result? This is probably known - the model sees batches with zero padding during training. However, the longest audio is not padded with zeros and therefore, the end may seem cut off a bit. I use manual zero padding of the longest item [in the inference script](https://github.com/janvainer/speedyspeech/blob/8a6c42b8812b64b2c44a2a0b6766d284f7269824/code/inference.py#L76) to eliminate the issue and it semed to help in most of the cases. It is possible that there is some other issue. Did you use the inference script for your experiment?

**Comment by eqikkwkp25-cyber** [comment]
Sure, here we go 
[LongSentence](https://sndup.net/39pn)

**Comment by janvainer** [comment]
I see, thanks for posting this. Please make an issue on this with the appended audio. I'll look into it. Untill then, simply split your text on sentences :)
